### ISSUE TITLE
[Bug] - Next.js / React Query - `error` type is `unknown` and causes a Typescript error

### ISSUE THREAD

**Issue by rcwestlake** [zenstackhq/zenstack#1062]
**Description and expected behavior**
A clear and concise description of what the bug is and what's the expected behavior.

**Screenshots**
Unknown type:
![image](https://github.com/zenstackhq/zenstack/assets/9679076/589de9e2-0a16-40e1-a198-e89c39fbac2c)
The hook:
![image](https://github.com/zenstackhq/zenstack/assets/9679076/60ca2801-94fb-4bb3-b4d0-b6116d0e98cb)
Error:
![image](https://github.com/zenstackhq/zenstack/assets/9679076/fb04f9bf-7ac3-40cf-adfb-fd2efbf918f0)
If I switch to a ternary style, it still complains:
![image](https://github.com/zenstackhq/zenstack/assets/9679076/af6f48ed-5343-4ebe-9634-7d149241c1b7)

**Environment (please complete the following information):**

-   ZenStack version: [e.g., 1.0.0-alpha.50]
```
"@zenstackhq/runtime": "^1.9.0",
"@zenstackhq/server": "^1.9.0",
"@zenstackhq/tanstack-query": "^1.9.0",
"zenstack": "1.9.0"
```
-   Prisma version: `"^5.10.2"`
-   Database type: Postgresql

**Additional context**

**Comment by ymc9** [comment]
Thanks for reporting this @rcwestlake ! Yes, we should have better typing for the error. Will make a fix in the next release.

**Comment by ymc9** [comment]
Fixes in 1.10.0
